private/pkg/git/metadata.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"strings"
+
+	"github.com/bufbuild/buf/private/pkg/command"
+)
+
+const (
+	gitCommand      = "git"
+	gitOriginRemote = "origin"
+	tagsPrefix      = "refs/tags/"
+	headsPrefix     = "refs/heads/"
+)
+
+// CheckForUncommittedGitChanges checks if there are any uncommitted and/or unchecked
+// changes from git based on the given directory.
+func CheckForUncommittedGitChanges(
```

Comment:
Why were these functions split out into a separate file than `git.go`, but `Remote` was not?

Reply:
Oh yeah, I put this in a separate file initially because these didn't come with an additional interface abstraction. But I've moved them to `git.go`, since it does conform with the old Go package styles.

Reply:
Oh yeah, I put this in a separate file initially because these didn't come with an additional interface abstraction. But I've moved them to `git.go`, since it does conform with the old Go package styles.